### DISCUSSION TITLE
Replacements/updates/enhancements

### DISCUSSION THREAD

**Discussion by rtritto** [mongo-express/mongo-express#942] [Ideas]
Some core replacement:
- Express.js, CodeMirror with web framework like Next.js, Nest, Fastify etc
- Bootstrap with Material UI, Tailwind CSS, Mantine etc
- JQuery with React, Vue, Solid.js etc
- ~yarn v1 with yarn berry~

We can extend the project with Electron to use it cross OS without Docker engine.

I know and use Next.js, Material UI, React.

**Comment by rtritto** [comment]
FYI @shakaran @BlackthornYugen @dozoisch @MahdiAbbasi95

**Comment by shakaran** [comment]
Before to address new core replacements, I will purpose get a Pull Request counter near to zero, and issues counter near to zero. When we resolve a lot of old issues, then change to new things. Some of them, you already start to implement it like yarn berry, so you can progress if you want meanwhile we fix all the known issues in the repo.

**Reply by xTudoS** [reply]
I agree with you, currently mongo-express is simple to maintenance so I, personally, don't see any vantages to use a powerful JS framework for UI. And talking about css frameworks (bootstrap, material ui, tailwind css), I think this could be implemented as alternatives themes in the project.

**Comment by rtritto** [comment]
A solution where I did some tests:
- [Solid.js](https://www.solidjs.com) (similar to [React](https://react.dev) but faster)
- [VPS (vite-plugin-ssr)](https://vite-plugin-ssr.com) (based on [Next.js](https://nextjs.org) but with island, native support of SSR)
- [SUID](https://suid.io) (port of [Material-UI](https://mui.com)):
  - some components like Select have problems with SSR
  - components like Pagination not yet released
- [solid-jotai](https://github.com/wobsoriano/solid-jotai) (port of [Jotai](https://jotai.org) state manager)

Maybe replace SUID with [Zag](https://zagjs.com) (less components atm).

Repository: https://github.com/rtritto/mongo-vps

**Comment by rtritto** [comment]
As stable ecosystem we can use:
- [React](https://react.dev)
- [Next.js](https://nextjs.org)
- [Material-UI](https://mui.com)
- [Jotai](https://jotai.org)

I have already started: https://github.com/rtritto/mongo-express-pwa

**Comment by rtritto** [comment]
Added repositories

**Comment by rtritto** [comment]
Replaced by #1705
